libmattermost.c

```diff
@@ -970,6 +970,65 @@ mm_add_user_to_channel_response(MattermostAccount *ma, JsonNode *node, gpointer
 	mm_join_room(ma, channel);
 }
 
+// when Pidgin is open and a new channel is joined then add it to the blist
+static void
+mm_add_new_joined_channel_to_blist(MattermostAccount *ma, MattermostChannel *mm_channel)
+{
+	if (mm_channel_is_hidden(ma, mm_channel->id)) {
+		mm_g_free_mattermost_channel(mm_channel);
+	} else {

```

Comment:
Add a return in the `if` block and remove the `else` block. Always try to keep if/else blocks as small as possible for better readability.